### PR TITLE
perf: Provisioner changes trigger consolidation to be re-considered

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes # 

**Description**

- Allows consolidation to automatically be triggered if there is a change to the provisioner spec
- This allows immediate reconciliation when consolidation is turned on or when there is a change to provisioner requirements

**How was this change tested?**

* Manual validation of E2E
* `make test`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note
None
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/state/provisioner.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	// If the provisioner is deleted, no reason to re-consider consolidation at this point
+	if err := c.kubeClient.Get(ctx, req.NamespacedName, stored); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+	// Something changed in the provisioner so we should re-consider consolidation
+	c.cluster.recordConsolidationChange()
```
Comment (tzneal) [inline pkg/controllers/state/provisioner.go:59]: Have you added a log to see how often this is called?  There's a counter controller that updates provisioner status which is triggered off node updates, which I think occur once a minute.

Comment (jonathan-innis) [reply pkg/controllers/state/provisioner.go:59]: Provisioner status won't trigger this reconciliation loop, only `.spec` updates (see the `GenerationChangedPredicate` below)

Comment (tzneal) [reply pkg/controllers/state/provisioner.go:59]: Ah, nice I missed that!

**Review by tzneal — APPROVED** [review]
lgtm

**Review by tzneal — APPROVED** [review]
lgtm

File: pkg/controllers/state/provisioner.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	return reconcile.Result{}, nil
+}
+
+func (c *ProvisionerController) Register(ctx context.Context, m manager.Manager) error {
+	return controllerruntime.
+		NewControllerManagedBy(m).
+		Named(provisionerControllerName).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 10}).
```
Comment (njtran) [inline pkg/controllers/state/provisioner.go:56]: Do we need max reconciles to be 10 here or are you just following pattern from other controllers?

File: pkg/controllers/state/provisioner.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		WithEventFilter(predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
```
Comment (njtran) [inline pkg/controllers/state/provisioner.go:58]: You could consider using the vended function for this here

https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/predicate/predicate.go#L156-L172

**Comment by njtran** [comment]
Looks like i was a little late :(

File: pkg/controllers/state/provisioner.go (hunk 2 above)
Comment (jonathan-innis) [reply pkg/controllers/state/provisioner.go:56]: Following the same pattern, not sure why we would have an issue with concurrent reconciles here.

**Pull request merged**